### PR TITLE
Wip/mos 83 onboarding page

### PR DESCRIPTION
**Pull request by naufaldirfq**
# Description
Create sliding onboarding page

# Ticket
https://berkatjayagroup.atlassian.net/browse/MOS-86

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: Moku/Services/AppState.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+//  AppState.swift
+//  Moku
+//
+//  Created by Naufaldi Athallah Rifqi on 25/10/21.
+//
+
+import SwiftUI
+
+class AppState: ObservableObject {
+    @Published var hasOnboarded: Bool
```
Comment (chrsomle) [inline Moku/Services/AppState.swift]: bisa bgt langsung pakein userdefaults: @Published-nya diganti @AppStorage, kyk gini aja nih:
`@AppStorage(UserDefaults.hasOnboarded) var hasOnboarded = false`

File: Moku/Services/AppState.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    init(hasOnboarded: Bool) {
```
Comment (chrsomle) [inline Moku/Services/AppState.swift]: karena udh dikasih default value `false` kyknya si init dikosongin sabi sih

File: Moku/Views/Onboarding/OnboardingDataModel.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//
+//  OnboardingDataModel.swift
+//  Moku
+//
+//  Created by Naufaldi Athallah Rifqi on 25/10/21.
+//
+
+import Foundation
+import UIKit
```
Comment (chrsomle) [inline Moku/Views/Onboarding/OnboardingDataModel.swift]: import uikit buat apa btw

File: Moku/Views/Onboarding/OnboardingStepView.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import SwiftUI
+
+struct OnboardingStepView: View {
+    
+    var data: OnboardingDataModel
+    
+    var body: some View {
+        VStack(alignment: .center) {
+            Text(data.image).font(.title)
+            Text(data.heading).font(.title3).multilineTextAlignment(TextAlignment.center).foregroundColor(Color("PrimaryColor")).padding(.top, 300).padding(.bottom)
```
Comment (chrsomle) [inline Moku/Views/Onboarding/OnboardingStepView.swift]: di enter-enter pak, jadiin multiline biar lebi readable

File: Moku/Views/Onboarding/OnboardingStepView.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            Text(data.text).font(.body).multilineTextAlignment(TextAlignment.center)
```
Comment (chrsomle) [inline Moku/Views/Onboarding/OnboardingStepView.swift]: `.multilineTextAlignment(TextAlignment.center)`
bisa dijadiin
`.multilineTextAlignment(.center)` aja

File: Moku/Views/Onboarding/OnboardingStepView.swift [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+import SwiftUI
+
+struct OnboardingStepView: View {
+    
+    var data: OnboardingDataModel
+    
+    var body: some View {
+        VStack(alignment: .center) {
```
Comment (chrsomle) [inline Moku/Views/Onboarding/OnboardingStepView.swift]: VStack tuh default alignmentnnya udh .center, jd bisa diilangin; lgsg jadiin `VStack { ... }` aja

File: Moku/Views/Onboarding/OnboardingStepView.swift (hunk 4 above)
Comment (chrsomle) [reply Moku/Views/Onboarding/OnboardingStepView.swift]: kyknya `Color("PrimaryColor")` udh sering bgt muncul, bikin `AppColor.swift` di folder Constants aja kali ya?

File: Moku/Views/Onboarding/OnboardingView.swift [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                .frame(maxWidth: .infinity)
+                .background(Color("PrimaryColor"))
+                .foregroundColor(.white)
+                .clipShape(RoundedRectangle(cornerRadius: 5.0))
+                .padding(.horizontal)
+            Button("Masuk ke halaman utama", action:{})
+                .padding()
+                .frame(maxWidth: .infinity)
+                .background(Color("SalmonOrange"))
+                .foregroundColor(Color("PrimaryColor"))
```
Comment (chrsomle) [inline Moku/Views/Onboarding/OnboardingView.swift:46]: nih di file ini muncul byk bgt

File: Podfile.lock [hunk 8]
```diff
@@ -21,4 +21,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: e6848d4974beb1250d94fd51a6a370683d14f196
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.10.2
```
Comment (chrsomle) [inline Podfile.lock]: cocoapodsnya di lu downgrade nih, kyknya perlu lu update bang

File: Moku/MokuApp.swift [hunk 9]
```diff
@@ -65,17 +70,22 @@ struct MokuApp: App {
...
                 }
-                
-                Text("Anda harus login.")
-                    .onAppear {
-                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
-                            let customer = Customer(name: "Alpha", phoneNumber: "1234")
-                            session.user = .customer(customer)
-                        }
-                    }
+//                GoogleMapView(coordinate: $userLocation.coordinate) {
```
Comment (chrsomle) [inline Moku/MokuApp.swift:78]: ohiya maap, ni commentsnya ntr gw yg beresin di branchh gw

Comment (chrsomle) [reply Moku/MokuApp.swift:78]: belom di rapiin

**Review by chrsomle — CHANGES_REQUESTED** [review]

File: Moku/Services/AppState.swift (hunk 1 above)
Comment (chrsomle) [reply Moku/Services/AppState.swift]: btw itu gw bikin `UserDefaults.swift` di folder Constants, gini:
`
struct UserDefaults {
    static let hasOnboarded = "has_onboarded"
}
`

File: Moku/Views/Onboarding/OnboardingStepView.swift (hunk 4 above)
Comment (naufaldirfq) [reply Moku/Views/Onboarding/OnboardingStepView.swift]: > kyknya `Color("PrimaryColor")` udh sering bgt muncul, bikin `AppColor.swift` di folder Constants aja kali ya?

ini katanya @albertusfavian udah bikin di branch lain. ya kan ya? wkwkwk

File: Moku/Views/Onboarding/OnboardingDataModel.swift (hunk 3 above)
Comment (naufaldirfq) [reply Moku/Views/Onboarding/OnboardingDataModel.swift]: kepencet

File: Moku/Views/Onboarding/OnboardingView.swift (hunk 7 above)
Comment (naufaldirfq) [reply Moku/Views/Onboarding/OnboardingView.swift:46]: kyknya dibikin reusable component aja kali ya?

File: Moku/Services/AppState.swift (hunk 1 above)
Comment (naufaldirfq) [reply Moku/Services/AppState.swift]: > btw itu gw bikin `UserDefaults.swift` di folder Constants, gini:
> `struct UserDefaults { static let hasOnboarded = "has_onboarded" }`

gua gak nemu filenya nih. ini di branch mana ya kris?

Comment (chrsomle) [reply Moku/Services/AppState.swift]: ohh emg blm pal, lu yg bikin aha

**Review by chrsomle — APPROVED** [review]

File: Moku/Views/Onboarding/OnboardingView.swift (hunk 7 above)
Comment (chrsomle) [reply Moku/Views/Onboarding/OnboardingView.swift:46]: eh mksd gw si Color(...) yg munculnya sering bgt

Comment (chrsomle) [reply Moku/Views/Onboarding/OnboardingView.swift:46]: masukin ke AppColor

File: Moku/Views/Onboarding/OnboardingStepView.swift (hunk 4 above)
Comment (chrsomle) [reply Moku/Views/Onboarding/OnboardingStepView.swift]: Kalo udh, merge sabi sih ntr paling dibenerin di branch si albert

**Comment by chrsomle** [comment]
eh keren bgt btw ada checklistnya skrg

**Pull request merged**